agent/consul/server.go

```diff
@@ -2237,22 +2238,70 @@ func (s *Server) hcpServerStatus(deps Deps) hcp.StatusCallback {
 
```

Comment:
I think we should break this function up to make the new changes easier to test

What if we create a new function like `populateServerTLS(status *hcpclient.ServerStatus, tlsConfigurator *tlsutil.Configurator) error`? That could encompass all the logic from L2238 - L2305

The other stuff happening in this status function (autopilot/raft) is more cumbersome since it requires a server, so this would just be an incremental improvement

Reply:
Thanks for the suggestion! I'll take a pass at this and let you know how it goes.

Reply:
Let me know what you think:  https://github.com/hashicorp/consul/pull/19682/commits/06542c62bbf3ef77764b65cf50397cb0d9ff63bd